Week 3/Extra Practice Problems (Week 3)/0087: Scramble String/dominic.cpp

```diff
@@ -0,0 +1,34 @@
+class Solution {
+public:
+    vector<vector<vector<int>>> memo;
+
+    bool solve(string s1, string s2, int l, int r) {  
```

Comment:
We should avoid adding extra variables in dfs call stack .


All we need really is l,r, and their length